core/state/database.go

```diff
@@ -132,11 +132,26 @@ func NewDatabase(db ethdb.Database) Database {
 	return NewDatabaseWithConfig(db, nil)
 }
 
+var (
+	dbKeyOverlayTransitionStarted = []byte("overlay-transition-started")
+	dbKeyOverlayTransitionEnded   = []byte("overlay-transition-ended")
+)
+
 // NewDatabaseWithConfig creates a backing store for state. The returned database
 // is safe for concurrent use and retains a lot of collapsed RLP trie nodes in a
 // large memory cache.
 func NewDatabaseWithConfig(db ethdb.Database, config *trie.Config) Database {
 	csc, _ := lru.New(codeSizeCacheSize)
+
+	transitionStarted, err := db.Has(dbKeyOverlayTransitionStarted)
+	if err != nil {
+		panic(err)
```

Comment:
Adding a `error` return in `NewDatabaseWithConfig` seems like a big step since it has an impact on many call sites.

Not that's strictly a problem, but I'd like your input on this matter before some "exploding" change stream. I'm not convinced of this panic, but leaving it now just to save our time while doing benchmarks temporarily.

If we want to avoid adding an error parameter in this method, feels like the start/end markers should be passed by callers (e.g: in existing parameters, ne